pkg/solana/chain_test.go

```diff
@@ -371,3 +318,305 @@ func TestChain_Transact(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, fees.ComputeUnitLimit(500), limit)
 }
+
+func TestSolanaChain_MultiNode_GetClient(t *testing.T) {
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		out := fmt.Sprintf(TestSolanaGenesisHashTemplate, client.MainnetGenesisHash) // mainnet genesis hash
+		if !strings.Contains(r.URL.Path, "/mismatch") {
+			// devnet gensis hash
+			out = fmt.Sprintf(TestSolanaGenesisHashTemplate, client.DevnetGenesisHash)
+		}
+		_, err := w.Write([]byte(out))
+		require.NoError(t, err)
+	}))
+	defer mockServer.Close()
+
+	ch := solcfg.Chain{}
+	ch.SetDefaults()
+	mnCfg := solcfg.MultiNodeConfig{
+		MultiNode: solcfg.MultiNode{
+			Enabled: ptr(true),
+		},
+	}
+	mnCfg.SetDefaults()
+
+	cfg := &solcfg.TOMLConfig{
+		ChainID:   ptr("devnet"),
+		Chain:     ch,
+		MultiNode: mnCfg,
+	}
+	cfg.Nodes = []*solcfg.Node{
+		{
+			Name: ptr("devnet"),
+			URL:  config.MustParseURL(mockServer.URL + "/1"),
+		},
+		{
+			Name: ptr("devnet"),
+			URL:  config.MustParseURL(mockServer.URL + "/2"),
+		},
+	}
+
+	testChain, err := newChain("devnet", cfg, nil, logger.Test(t))
+	require.NoError(t, err)
+
+	err = testChain.Start(tests.Context(t))
+	require.NoError(t, err)
+	defer func() {
+		closeErr := testChain.Close()
+		require.NoError(t, closeErr)
+	}()
+
+	selectedClient, err := testChain.getClient()
+	assert.NoError(t, err)
+
+	id, err := selectedClient.ChainID(tests.Context(t))
+	assert.NoError(t, err)
+	assert.Equal(t, "devnet", id.String())
+}
+
+func TestChain_MultiNode_TransactionSender(t *testing.T) {
+	ctx := tests.Context(t)
+	url := client.SetupLocalSolNode(t)
+	lgr, _ := logger.TestObserved(t, zapcore.DebugLevel)
+
+	// transaction parameters
+	sender, err := solana.NewRandomPrivateKey()
+	require.NoError(t, err)
+	receiver, err := solana.NewRandomPrivateKey()
+	require.NoError(t, err)
+	client.FundTestAccounts(t, solana.PublicKeySlice{sender.PublicKey()}, url)
+
+	// configuration
+	cfg := solcfg.NewDefault()
+	cfg.MultiNode.MultiNode.Enabled = ptr(true)
+	cfg.Nodes = append(cfg.Nodes,
```

Comment:
each of the nodes here is a separate network. This could lead to a flaky behaviour or confusion in the future